### PR TITLE
Category Api Created For Adding And Editing(server Side)

### PR DESCRIPTION
**Pull request by spartha1995**
File Changed: Startup.cs
ICategoriesRepository
CategoriesRepository
CategoryController

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/Categories/CategoryRepository.cs [hunk 1]
```diff
@@ -21,5 +23,48 @@ public IEnumerable<string> GetAllCategories()
...
+
+        #region Finding a Id Respective Category
+        /// <summary>
+        /// Find a Respective Id from Catagory Table
+        /// </summary>
+        /// <param name="Key"></param>
+        /// <Returns>if key foundthen Return respective category from category table or will return Null</Returns>
+        public Category Getcategory(int key)
+        {
+            try
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Categories/CategoryRepository.cs]: what is the requirement of try catch here?

File: Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs [hunk 2]
```diff
@@ -21,6 +22,53 @@ public IActionResult GetAllCategories()
...
+        /// <param name="catagory">Object of  class Category</param>
+        /// <returns>object of the class if key found or else it will return Bad request</returns>
+        [HttpPut("{id}")]
+        public IActionResult CategoryEdit(int Id, [FromBody] Category category)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            var promise = _categoryRepository.Getcategory(Id);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs]: the name is not proper

Comment (spartha1995) [inline Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs]: which name??

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs]: variable name is not proper.

File: Trappist/src/Promact.Trappist.Repository/Categories/CategoryRepository.cs [hunk 3]
```diff
@@ -21,5 +23,43 @@ public IEnumerable<string> GetAllCategories()
...
+        #endregion
+
+        #region Finding a Id Respective Category
+        /// <summary>
+        /// Find a Respective Id from Catagory Table
+        /// </summary>
+        /// <param name="Key"></param>
+        /// <Returns>if key foundthen Return respective category from category table or will return Null</Returns>
+        public Category Getcategory(int key)
+        {
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Categories/CategoryRepository.cs]: please do proper formatting.

**Pull request merged**